core/app_streaks_notifications.go

```diff
@@ -0,0 +1,299 @@
+func (n streaksNotifications) processStreaks() {
+	now := time.Now().UTC()
+	nowSeconds := 60*60*now.Hour() + 60*now.Minute() + now.Second()
+
+	courseConfigs, err := n.storage.FindCourseConfigs(nil)
+	if err != nil {
+		n.logger.Errorf("processStreaks -> error finding active course configs: %v", err)
+		return
+	}
+	if len(courseConfigs) == 0 {
+		n.logger.Errorf("processStreaks -> no active course configs for streaks")
+		return
+	}
+
+	// completedtask
+	// when complete usercourseunitprogress, get utc convert to local using timezone name and offset, store date
+	// when we revist in process streaks, if same date, we ignore. if yesterday then increment
+
+	// for every config in course_configs:
+	// 		retrieve usercourse that passes 12am local time (how?) (also what if china to usa, experiencing same day mightnight twice)
+	//		if CompletedTasks? (refer to comments above):
+	//			streaks + 1; pause + 1
+	//			if usercourse's new pause > max_pause:
+	//				set to max_pause
+	//		else (incomplete task):
+	//			pause - 1
+	//			if pause < 0:
+	//				streak = 0; pause = 0
+	//		updateUserCourses([usercourses struct]) (why db usercourse doesn't have timezone?)
+	//		reset CompletedTasks?
+	// remember to assign initial pause when creating usercourse struct
+
+	for _, config := range courseConfigs {
+		tzOffsets := make(model.TZOffsets, 0)
+		var userCourses []model.UserCourse
+		// 0 for streak processtime
+		offset := 0 - nowSeconds
+		if config.StreaksNotificationsConfig.TimezoneName == "user" {
+			if offset >= minTZOffset && offset <= maxTZOffset {
+				tzOffsets = append(tzOffsets, offset)
+			}
+
+			if offset+utils.SecondsInDay <= maxTZOffset {
+				tzOffsets = append(tzOffsets, offset+utils.SecondsInDay)
+			}
+			if offset-utils.SecondsInDay >= minTZOffset {
+				tzOffsets = append(tzOffsets, offset-utils.SecondsInDay)
+			}
+
+			// load user courses for this course based on timezone offsets
+			userCourses, err = n.storage.FindUserCourses(nil, config.AppID, config.OrgID, nil, []string{config.CourseKey}, nil, tzOffsets.GeneratePairs(config.StreaksNotificationsConfig.PreferEarly), nil)
+			if err != nil {
+				n.logger.Errorf("processStreaks -> error finding user courses for course key %s: %v", config.CourseKey, err)
+				continue
+			}
+		} else {
+			configOffset := config.StreaksNotificationsConfig.TimezoneOffset
+			if offset == configOffset || offset+utils.SecondsInDay == configOffset || offset-utils.SecondsInDay == configOffset {
+				// load all user courses for this course
+				userCourses, err = n.storage.FindUserCourses(nil, config.AppID, config.OrgID, nil, []string{config.CourseKey}, nil, nil, nil)
+				if err != nil {
+					n.logger.Errorf("processStreaks -> error finding user courses for course key %s: %v", config.CourseKey, err)
+					continue
+				}
+			}
+		}
+
+		for i, userCourse := range userCourses {
+			if userCourse.CompletedTasks != nil {
+				uY, uM, uD := userCourse.CompletedTasks.Date()
+				y, m, d := now.Date()
```

Comment:
I think this needs to be converted to the user's timezone using `time.FixedZone`, otherwise it will always use the timezone where the system running the LMS BB is.